packages/paste-website/src/components/customization-landing-page/image-slider/ImageSlider.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import type {MutableRefObject, LegacyRef} from 'react';
+import {Box} from '@twilio-paste/box';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+
+import type {FluidObject} from 'gatsby-image';
+import {SVGThumb} from './SVGThumb';
+import {ImageBox} from './ImageBox';
+import {DEFAULT_MIN_CHANGE, MAX_VALUE, MIN_VALUE} from './constants';
+import {convertPositionToInputValue, clampValueToRange} from './utils';
+import {useSvgResize} from './useSvgResize';
+
+const INPUT_ID = 'input-range-id';
+const CLIP_PATH_ID = 'clip-path-id';
+
+export const ImageSlider: React.FC<{frontFluidObject: FluidObject; backFluidObject: FluidObject}> = ({
+  frontFluidObject,
+  backFluidObject,
+}) => {
+  const uidSeed = useUIDSeed();
+
+  // Set the initial state of the slider to be roughly at the 60% position.
+  const [value, setValue] = React.useState<number>(MAX_VALUE * 0.6);
+  const [isDragging, setIsDragging] = React.useState(false);
+  const [refsAreInitiated, setRefsAreInitiated] = React.useState(false);
+
+  const containerRef = React.useRef<HTMLElement>();
+  const svgCircleRef = React.useRef<SVGCircleElement>();
+
+  const {width: containerWidth, x: containerX, height: containerHeight} = containerRef?.current
+    ? containerRef.current.getBoundingClientRect()
+    : {width: 0, height: 0, x: 0};
+
+  const {svgOffset, svgHeight, svgWidth} = useSvgResize(containerHeight, containerWidth);
+
+  // Minimum change defines the absolute distance threshold for value to change.
+  // Calculated by half the width of the SVG, default is half of the absolute width.
+  const minimumChange = svgWidth ? svgWidth / 2 : DEFAULT_MIN_CHANGE;
+
+  // Width of SVG Clip used to conditionally show/hide part of the each image
+  // SVG clip is calculated from right to left, since the dynamic value here is the width.
+  // const clip = (value / MAX_VALUE) * containerWidth;
+  const clip = (value / MAX_VALUE) * containerWidth;
+
+  const handleMouseMove = React.useCallback(
```

Comment:
Could you explain the decision to memoize these but not the `onMouseDown` handler on line 103?

Reply:
I would say it was a _lack_ of a decision :P

Reply:
Ha ok, I'll leave them for now since I feel like it's pretty negligible. I am going too remove the commented out line 42 also, but then that is my last update. 🥳